pypeit/calibrations.py

```diff
@@ -915,23 +976,225 @@ def __repr__(self):
         txt += '>'
         return txt
 
+    @staticmethod
+    def get_association(fitstbl, spectrograph, caldir, setup, calib_ID, det, must_exist=True,
```

Comment:
this is nice!

Reply:
Is there a test on this?
  if not, please add one

Reply:
Thanks!  Yes, there are two tests in `test_calibrations.py`.